File: api/src/main/java/net/blueshell/api/base/BaseDTO.java
package net.blueshell.api.base;

import lombok.Data;

import java.io.Serializable;
import java.time.Instant;

@Data
public abstract class BaseDTO implements Serializable {
    private Long id;
    private Instant deletedAt;
    private Instant createdAt;
    private Instant updatedAt;
    private Long version;
}

File: api/src/main/kotlin/net/blueshell/api/base/BaseDTO.kt
package net.blueshell.api.base

import java.io.Serializable
import java.time.Instant

abstract class BaseDTO : Serializable {
    open var id: Long? = null
    open var deletedAt: Instant? = null
    open var createdAt: Instant? = null
    open var updatedAt: Instant? = null
    open var version: Long? = null
}
